syndesis/src/modules/data/pages/VirtualizationCreatePage.tsx

```diff
@@ -1,21 +1,116 @@
 import { Breadcrumb, Container } from '@syndesis/ui';
+import { WithRouteData } from '@syndesis/utils';
 import * as React from 'react';
+import { Translation } from 'react-i18next';
 import { Link } from 'react-router-dom';
-import resolvers from '../resolvers';
+import i18n from '../../../i18n';
+import resolvers from '../../resolvers';
 
 export default class VirtualizationCreatePage extends React.Component {
   public render() {
+    const formDefinition = {
+      virtDescription: {
+        componentProperty: true,
+        deprecated: false,
+        displayName: i18n.t(
+          'data:virtualization.virtualizationDescriptionDisplay'
+        ),
+        javaType: 'java.lang.String',
+        kind: 'property',
+        order: 1,
+        required: false,
+        secret: false,
+        type: 'textarea',
+      },
+      virtName: {
+        componentProperty: true,
+        deprecated: false,
+        displayName: i18n.t('data:virtualization.virtualizationNameDisplay'),
+        javaType: 'java.lang.String',
+        kind: 'property',
+        order: 0,
+        required: true,
+        secret: false,
+        type: 'string',
+      },
+    } as IFormDefinition;
+    const validate = (v: { virtName: string }) => {
+      const errors: any = {};
+      // TODO Incorporate service call to validate virtualization name
+      if (v.virtName.includes('?')) {
+        errors.virtName = 'Virtualization name contains an illegal character';
+      }
+      return errors;
+    };
+
     return (
-      <>
-        <Breadcrumb>
-          <Link to={resolvers.root()}>Data Virtualizations</Link>
-          <span>Create Virtualization</span>
-        </Breadcrumb>
-        <Container>
-          <h1>Create Virtualization</h1>
-          <p>Components for creating the virtualization go here</p>
-        </Container>
-      </>
+      <WithRouteData<null, null>>
+        {(p, s, { history }) => (
+          // TODO need to retrieve real user here
+          <WithVirtualizationHelpers username="developer">
+            {({ createVirtualization }) => {
+              const handleCreate = async (value: any) => {
+                await createVirtualization(
+                  value.virtName,
+                  value.virtDescription
+                );
+                // TODO: post toast notification
+                history.push(resolvers.data.virtualizations.list());
+              };
+              return (
+                <Translation ns={['data', 'shared']}>
+                  {t => (
+                    <>
+                      <Breadcrumb>
+                        <Link to={resolvers.dashboard.root()}>
+                          {t('shared:Home')}
+                        </Link>
+                        <Link to={resolvers.data.root()}>
+                          {t('shared:DataVirtualizations')}
+                        </Link>
+                        <span>
+                          {t('shared:Create')} {t('shared:DataVirtualization')}
```

Comment:
You should reference one i18n key here. Then in the localization file reference other keys. That gives the language the choice to order them or not use them if they choose.

Reply:
ok - I'll change that